uniflow/op/model/model_server.py

```diff
@@ -769,3 +770,223 @@ def __call__(self, data: List[str]) -> List[str]:
+                "Please install it by running `pip install boto3`."
+            ) from e
+        except Exception as e:
+            raise ValueError(
+                "Failed to load credentials for authenticating with the AWS client. "
+                "Please ensure that the specified profile name contains valid credentials."
+            ) from e
+
+    def _preprocess(self, data: List[str]) -> List[str]:
+        """Preprocess data.
+
+        Args:
+            data (List[str]): Data to preprocess.
+
+        Returns:
+            List[str]: Preprocessed data.
+        """
+        return data
+
+    def _postprocess(self, data: List[str]) -> List[str]:
+        """Postprocess data.
+
+        Args:
+            data (str): Data to postprocess.
+
+        Returns:
+            List[str]: Postprocessed data.
+        """
+        return data
+
+    def enforce_stop_tokens(self, text: str, stop: List[str]) -> str:
+        """Cut off the text as soon as any stop words occur."""
+        return re.split("|".join(stop), text, maxsplit=1)[0]
+
+    def prepare_input(
+        self, model_type: str, prompt: str, model_kwargs: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """
+        Prepare the input for the model based on the model_type.
+
+        Args:
+            model_type (str): The type of the model.
+            prompt (str): The input prompt.
+            model_kwargs (Dict[str, Any]): Additional model arguments.
+
+        Returns:
+            Dict[str, Any]: The prepared input for the model.
+        """
+
+        def prepare_falcon_input(
+            prompt: str, model_kwargs: Dict[str, Any]
+        ) -> Dict[str, Any]:
+            input_body = {
+                "inputs": f"{prompt}",
+                "parameters": model_kwargs,
+            }
+            return json.dumps(input_body)
+
+        def prepare_default_input(
+            prompt: str, model_kwargs: Dict[str, Any]
+        ) -> Dict[str, Any]:
+            input_str = json.dumps({"inputs": prompt, "parameters": model_kwargs})
+            return input_str
+
+        model_input_preparation = {"falcon": prepare_falcon_input}
+
+        prepare_input_for_model = model_input_preparation.get(
+            model_type, prepare_default_input
+        )
+        return prepare_input_for_model(prompt, model_kwargs)
+
+    def prepare_output(self, model_type: str, response: Any) -> str:
+        """
+        Prepares the output based on the model_type and response.
+
+        Args:
+            model_type (str): The model_type of the response.
+            response (Any): The response object.
+
+        Returns:
+            str: The prepared output.
+
+        Raises:
+            None
+        """
+
+        def prepare_default_output(response: Any) -> str:
+            response_body = json.loads(response.get("Body").read())
+            return response_body[0].get("generated_text")
+
+        model_output_preparation = {"falcon": prepare_default_output}
```

Comment:
nit: are we using falcon as the default model? If so, can we change to mistral model as default?